Handle arbitrary templates in post_from_url

cschorn opened the pull request:
`post_from_url` was to dependent on slashes separating single
attributes. Permalink templates like `(year)-(month)/(url)` would
fail. This is corrected by not trying to normalize the request URL
or the template URL when building the regex.

cschorn committed:
Handle arbitrary templates in post_from_url

`post_from_url` was to dependent on slashes separating single
attributes. Permalink templates like `(year)-(month)/(url)` would
fail. This is corrected by not trying to normalize the request URL
or the template URL when building the regex.

xhezairbey commented:
Looks like a great change in the URL template, but I'm just curios whether this was already tested before I merge it.
Thanks.

cschorn commented:
Only by hand so far (I somehow had overlooked `tester.rb`). I'll try to get it to run and get back to you.

cschorn commented:
OK, tests worked although `tester.rb` seems awfully buggy (was missing the security key for creating posts, for example).

I'll have another pull request ready in a few hours, I guess. :)

cschorn committed:
Updated tests to use `secure_hashkey` from config

When running `tester.rb`, the tests for adding posts and pages failed
because the `hash` parameter was missing from the post data.
